src/ucp/core/ucp_rkey.c

```diff
@@ -451,19 +454,20 @@ void ucp_rkey_resolve_inner(ucp_rkey_h rkey, ucp_ep_h ep)
 {
     ucp_context_h context   = ep->worker->context;
     ucp_ep_config_t *config = ucp_ep_config(ep);
+    int rma_sw              = 0;
+    int amo_sw              = 0;
     ucs_status_t status;
     uct_rkey_t uct_rkey;
-    int rma_sw, amo_sw;
 
     rkey->cache.rma_lane = ucp_rkey_find_rma_lane(context, config,
                                                   UCS_MEMORY_TYPE_HOST,
                                                   config->key.rma_lanes, rkey,
                                                   0, &uct_rkey);
-    rma_sw = (rkey->cache.rma_lane == UCP_NULL_LANE);
-    if (rma_sw) {
-        rkey->cache.rma_proto     = &ucp_rma_sw_proto;
+    if (rkey->cache.rma_lane == UCP_NULL_LANE) {
         rkey->cache.rma_rkey      = UCT_INVALID_RKEY;
         rkey->cache.max_put_short = 0;
+        rkey->cache.rma_proto     = &ucp_rma_sw_proto;
```

Comment:
why moved at the 469 line?
let's stay the same order as it was and in other places (RMA/AMO/AMO-SW)

Reply:
reverted, the rest of the code is needed to avoid remote lane resolve if RMA/AMO are not enabled by feature flags